### PR TITLE
ref(consumer): Add the ability to override storage topic names through settings

### PR DESCRIPTION
**Pull request by tkaemming**
This addresses the immediate issues raised in https://github.com/getsentry/snuba/pull/1551#discussion_r544642804 that prevented the multistorage consumer from being able to be used in our production configuration.

### CODE REVIEW

**Review by tkaemming — COMMENTED** [review]
Note that this doesn't put the consumer command line parameters on any specific deprecation path, that can be either done now or later.

**Review by fpacifici — COMMENTED** [review]
Please see my comment about the StorageKey enum. Otherwise no issue with this.

File: snuba/datasets/table_storage.py [hunk 1]
```diff
@@ -89,13 +68,72 @@ def get_commit_log_topic_spec(self) -> Optional[KafkaTopicSpec]:
...
 
+def build_kafka_topic_spec_from_settings(topic_name: str) -> KafkaTopicSpec:
+    return KafkaTopicSpec(
+        topic_name=topic_name,
+        partitions_number=settings.TOPIC_PARTITION_COUNTS.get(topic_name, 1),
+    )
+
+
+def build_kafka_stream_loader_from_settings(
+    storage_name: str,
```
[the remark this answers is not in the document]
Comment (tkaemming) [reply snuba/datasets/table_storage.py]: No reason. Changed here: e66fc5e.

File: snuba/datasets/storages/errors.py [hunk 2]
```diff
@@ -30,10 +30,11 @@
...
     query_processors=query_processors,
     query_splitters=query_splitters,
-    stream_loader=KafkaStreamLoader(
+    stream_loader=build_kafka_stream_loader_from_settings(
+        StorageKey.ERRORS.name,
         processor=ErrorsProcessor(promoted_tag_columns),
-        default_topic="events",
-        replacement_topic="errors-replacements",
+        default_topic_name="events",
+        replacement_topic_name="errors-replacements",
```
[the remark this answers is not in the document]
Comment (tkaemming) [reply snuba/datasets/storages/errors.py:37]: I didn't see a benefit or justification to making that change and moving these stream definitions further from the storage definition, so I didn't make it.

`STORAGE_TOPICS` doesn't need to be copied, regardless of whether or not the defaults are set there or somewhere else. The value of a single top (storage) or second level (topic) key can be mutated without affecting the others as a normal dictionary.

**Review by fpacifici — APPROVED** [review]

**Pull request merged**